bgpd/bgp_vty.c

```diff
@@ -14622,7 +14626,7 @@ static void bgp_show_peer(struct vty *vty, struct peer *p, bool use_json,
 	}
 
 	/* Local address. */
-	if (p->su_local) {
+	if (p->su_local && (p->status == Established)) {
```

Comment:
Why would we want to change the show output for the non-json use case.  This will break people's scripts right?

Reply:
I think we should at least keep the results in various formats aligned. Or It will bring doubts or confusions.

Reply:
I've no idea about testing scripts on this.

In the previous logic, Local host, Local port in non-json output are conditional, only appear when (p->su_local). 
You mean scripts require to access them although status is not Established?
 
Could you advise in what conditions these fields are required to be present?
How about show invalid values (Unknown/-1) always like JSON in the else branch? 

Similar case for Foreign Host, Foreign port in the below comment.